tart2ms/tart2ms.py

```diff
@@ -135,8 +144,19 @@ def append(self, dataset):
         self.datasets.append(dataset)
 
     def write(self):
+        """ Creates a future to be computed 
+            ensure to call dask.compute with get_futures
+            to finalize graph construction
+        """
         writes = xds_to_table(self.datasets, self.table_name, columns="ALL")
-        dask.compute(writes)
+        self.__write_futures.append(writes)
+
+    @classmethod
+    def get_futures(cls):
+        """ returns a list of futures to be passed to dask.compute """
+        futures = cls.__write_futures
+        cls.__write_futures = []
+        return futures
```

Comment:
The reason for this change is so that we keep the graph execution to one point instead of multiple graph executions with associated overheads.